### PR TITLE
soc-ci: add artifacts download and workers pool overview

### PR DESCRIPTION
**Pull request by toabctl**
We are usually interessted in the supportconfigs when a
openstack-mkcloud job fails. So add the possibility to
download the Jenkins artifacts (which include the supportconfigs).

### CODE REVIEW

File: hostscripts/soc-ci/soc-ci [hunk 1]
```diff
@@ -250,6 +290,14 @@ def parse_args():
     parser_worker_pool_release.add_argument('slot', type=int)
     parser_worker_pool_release.set_defaults(func=ci_worker_pool_release)
 
+    # workers pool list - list *all* workers pool
+    parser_workers_pool_list = subparsers.add_parser(
+        'workers-pool-list', help='List CI workers pool for all workers')
+    parser_workers_pool_list.add_argument('--my-reservations',
+                                          action='store_true',
```
Comment (aspiers) [inline hostscripts/soc-ci/soc-ci]: Why not make this the default?

File: hostscripts/soc-ci/soc-ci [hunk 2]
```diff
@@ -126,6 +136,30 @@ def os_mkcloud_available(args):
...
+    if os.path.exists(dest_dir):
+        if not args.force:
+            print('Artifacts already available under %s' % dest_dir)
+            sys.exit(0)
+    else:
+        os.makedirs(dest_dir)
+
+    for filename, artifact in build.get_artifact_dict().items():
+        artifact.save_to_dir(dest_dir)
+    print('Artifacts downloaded to %s' % dest_dir)
```
Comment (aspiers) [inline hostscripts/soc-ci/soc-ci]: How about adding an option to run `unpack-supportconfig` on them after download?  Like in https://github.com/abel-navarro/cifetch/blob/master/cifetch.py but without reinventing wheels.

File: hostscripts/soc-ci/soc-ci (hunk 1 above)
Comment (toabctl) [inline hostscripts/soc-ci/soc-ci]: anyway - I changed it so it is now showing only the own reservations. There is now also an ```--all``` switch to show all .

File: hostscripts/soc-ci/soc-ci (hunk 2 above)
Comment (toabctl) [inline hostscripts/soc-ci/soc-ci]: @aspiers done

**Pull request merged**